zcutil/fetch-pyarams.py

```diff
@@ -0,0 +1,296 @@
+
+def verify_ipfs(filename, download_state):
+    
+    if HOST_OS == 'Windows':
+        p2 = Popen(['ipfs.exe', "add", "--n", "--Q", PARAMS_DIR + filename], stdout=PIPE)
+    else:
+        p2 = Popen(['ipfs', "add", "--n", "--Q", PARAMS_DIR + filename], stdout=PIPE)
+    p2.wait()
+    buffer2 = p2.communicate()[0]
+
+    if buffer2.decode("utf-8").replace('\n','') != PARAM_IPFS_CIDS.get(filename):
+        logger.error("Download failed: Multihash on %s does NOT match.", filename)
+        return
+    
+    if download_state == DOWNLOADING :
+        logger.info("Download successful!")
+        logger.info("%s: OK", filename)
+
+    if download_state == DOWNLOADED :
+        logger.info("%s: OK", filename)
+
+
+def verify_file(filename, sha256, download_state):
+    logger.debug("Checking SHA256 for: %s", filename) 
+    with open(filename, 'rb') as f:
+        try:
+    	    contents = f.read()
+        except:
+            logger.exception("Unable to read in data blocks to verify SHA256 for: %s", filename)
+        
+        local_sha256 =  hashlib.sha256(contents).hexdigest()
+        if local_sha256 != sha256 :
+            logger.error("Download failed: SHA256 on %s does NOT match.", filename)
+            return
+            
+    if download_state == DOWNLOADING :
+        logger.info("Download successful!")
+        logger.info("%s: OK", filename)
+        
+        try:
+            os.rename(filename, filename[:-3])
+        except:
+            logger.exception("Unable to rename file.")
+
+        logger.info("renamed '%s' -> '%s' \n", filename, filename[:-3])
+    
+    if download_state == DOWNLOADED :
+        logger.info("%s: OK", filename)
+
+def get_params(param_file_list, protocol):
+    for filename in param_file_list:
+        download_file(filename, protocol)
+
+def check_params(param_file_list, protocol):
+    for key in param_file_list:
+        if os.path.exists(PARAMS_DIR + key) == True and protocol == "HTTPS":
+            verify_file(PARAMS_DIR + key , param_file_list.get(key), DOWNLOADED )
+        elif os.path.exists(PARAMS_DIR + key) == True and protocol == "IPFS":
+            verify_ipfs(key , DOWNLOADED )
+        else :
+            logger.warning("%s does not exist and will now be downloaded...", PARAMS_DIR + key )
+            download_file( key, protocol)
+            
+def create_readme():
+    try:
+        #@TODO Update multi string
+        os.mkdir(PARAMS_DIR)
+        f = open(PARAMS_DIR + "README", "w+")
+        f.write("This directory stores common Zcash zkSNARK parameters. Note that it is \n")
+        f.write("distinct from the daemon's -datadir argument because the parameters are \n")
+        f.write("large and may be shared across multiple distinct -datadir's such as when \n")
+        f.write("setting up test networks.")
+        f.close()
+    except OSError as e:
+        logger.exception("Exception occured:", e)
+
+def print_intro():
+    print("Zcash - fetch-params.py \n")
```

Comment:
Is the file supposed to be named `fetch-pyarams.py` or `fetch-params.py`? I see different instances of this across the file.